src/sparseml/core/session.py

```diff
@@ -46,26 +56,53 @@ class _CallbackContainer:
 
 
 class SparseSession:
+    """
+    A session for sparsification that holds the current state of the sparsification
+    """
+
     def __init__(self):
         self._lifecycle = SparsificationLifecycle()
 
     @property
     def lifecycle(self) -> SparsificationLifecycle:
+        """
+        :return: the lifecycle for the session
+        """
```

Comment:
clarify that the lifecycle is used to keep track of where we are in the sparsification process